### PR TITLE
UInt parse Float and String

### PR DESCRIPTION
**Pull request by PXshadow**
Creates a String parser for UInt based upon Int64Helper and a Float parser created from scratch (not sure how sound it is) This is a pragmatic fix for #7535 at the current moment without support for larger max value integer literals.

### CODE REVIEW

**Comment by Aurel300** [comment]
Why are the two methods `private`? They are not called from other methods of `UInt`, they are not `@:from`/`@:to` marked, nor are they `@:op`s…

Regarding `parseFloat`, I don't think that's a good name for the method. Also I am not entirely sure what it is meant to accomplish. Is this meant to cover the case specifically when the source code contains an unsigned literal that fits in 32 bits (but the high bit is 1), so it is parsed as a `Float` and would need to be converted back into a `UInt`? Did you test the behaviour?

**Comment by PXshadow** [comment]
@Aurel300 Good point on the methods being ``private`` they are now ``public`` and I've removed parseFloat to remove the issues brought up.

File: std/UInt.hx [hunk 1]
```diff
@@ -307,6 +307,39 @@ abstract UInt(Int) from Int to Int {
 		return this;
 	}
 
+	public static #if (!js || analyzer) inline #end function parseString(s:String):UInt {
```
Comment (Simn) [inline std/UInt.hx]: Surely this function should not be `inline`.

**Comment by Simn** [comment]
This needs a test.

**Pull request closed without merging**